EIPS/eip-1654.md

```diff
@@ -0,0 +1,123 @@
+---
+eip: 1654
+title: Dapp-wallet authentication process with contract wallets support
+author: Maor Zamski (@pazams)
+discussions-to: https://github.com/ethereum/EIPs/issues/1654
+status: Draft
+type: Meta
+created: 2018-12-12
+---
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+An off-chain process for dapps to assert whether an entity has authorized control (informally, "ownership") over a public Ethereum address using `eth_sign`. Supports both external wallets and contract wallets.
+
+## Definitions
+- `contract wallet` A contract [account](https://github.com/ethereum/wiki/wiki/White-Paper#ethereum-accounts) deployed with the intent to be used as the ownership address for on-chain assets (including ether, ERC-20 tokens, and ERC-721 NFTs). It has the ability to transfer ether or dynamically execute actions on other contracts (acting as the owner of assets controlled by those contracts). Common examples of contract wallets are `multisig wallets` (such as the ones provided by [Gnosis](https://github.com/Gnosis/MultiSigWallet) and [Parity](https://github.com/ConsenSys/MultiSigWallet)) and `identity contracts`, as defined in [ERC-725](https://github.com/ethereum/EIPs/issues/725).
+- `external wallet` An externally owned [account](https://github.com/ethereum/wiki/wiki/White-Paper#ethereum-accounts), controlled by a private key. Currently, most on-chain assets are owned by such accounts. A common example for an external wallet are the wallets generated by MetaMask.
+- `authorized signer` An entity is considered to have authorized control over a wallet if either:
+  - It produced a signature of which the recovered address matches the wallet address.
+  - It produced a signature of which the contract at the wallet address responds with the magic value `0x1626ba7e` to the `IsValidSignature` contract call.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+The authentication process starts with the dapp client component requesting a message signature from the wallet.
+The client then proceeds to send the result to the dapp backend component along with the requested address to be used for authentication. The dapp backend recovers a public key from the signature, and checks if it has authorized control over the requested address. This check is done under consideration that the address may represent either an external wallet or a contract wallet. This process works with external wallets and contract wallets that support EIP-1271 with `0x1626ba7e` as a magic return value.
```

Comment:
Because of the mention of 1271, should this ERC depend on 1271?

Reply:
Good point 👍 
I cleared up the relation to 1271 to make it clearer it doesn't depend 1271.